source/common/config/filter_json.cc

```diff
@@ -238,6 +238,10 @@ void FilterJson::translateRedisProxy(
 
   auto* conn_pool = proto_config.mutable_settings();
   JSON_UTIL_SET_DURATION(*json_conn_pool, *conn_pool, op_timeout);
+
+  if (json_config.hasObject("downstream_auth_password")) {
```

Comment:
I think you can remove all the JSON parsing logic as [it's deprecated in v2](https://github.com/envoyproxy/envoy/pull/5623#discussion_r248459784).

Reply:
hmmm, I'll check into that. I think I added it for the config unit test...

Reply:
It's an integral part of proxy_filter_test.cc at the moment. The test calls translateRedisProxy() directly to create a RedisProxy proto config (in parseProtoFromJson()), so it's not using the deprecated FactoryBase::createFilterFactory() just some of its machinery...